src/request.spec.ts

```diff
@@ -482,375 +367,282 @@ describe('Request', () => {
+  it('repeats 429-throttled request after the designated retry value for large file (stream)', async () => {
     let i: number = 0;
     let timeout: number | undefined = -1;
+    const retryAfter = 60;
 
     sinon.stub(_request as any, 'req').callsFake(options => {
       _options = options as CliRequestOptions;
       (options as CliRequestOptions).responseType = "stream";
 
       if (i++ === 0) {
-        return Promise.reject({
+        throw {
           response: {
             status: 429,
             headers: {
               'retry-after': 60
             }
           }
-        });
+        };
       }
       else {
-        return Promise.resolve({ data: {} });
+        return { data: {} };
       }
     });
-    sinon.stub(global, 'setTimeout').callsFake((fn, to) => {
-      timeout = to;
-      fn();
-      return {} as any;
+
+    sinon.stub(_request as any, 'setTimeout').callsFake(async (value: any) => {
+      timeout = value;
+      return;
     });
 
-    _request
+    await _request
       .get({
         url: 'https://contoso.sharepoint.com/'
-      })
-      .then(() => {
-        try {
-          assert.strictEqual(timeout, 60000);
-          done();
-        }
-        catch (err) {
-          done(err);
-        }
-      }, (err) => {
-        done(err);
       });
+    assert.strictEqual(timeout, retryAfter);
   });
 
-  it('repeats 503-throttled request until it succeeds', (done) => {
+  it('repeats 503-throttled request until it succeeds', async () => {
     let i: number = 0;
 
     sinon.stub(_request as any, 'req').callsFake(() => {
       if (i++ < 3) {
-        return Promise.reject({
+        throw {
           response: {
             status: 503,
             headers: {}
           }
-        });
+        };
       }
       else {
-        return Promise.resolve({ data: {} });
+        return { data: {} };
       }
     });
-    sinon.stub(global, 'setTimeout').callsFake((fn) => {
-      fn();
-      return {} as any;
-    });
 
-    _request
+    await _request
       .get({
         url: 'https://contoso.sharepoint.com/'
-      })
-      .then(() => {
-        try {
-          assert.strictEqual(i, 4);
-          done();
-        }
-        catch (err) {
-          done(err);
-        }
-      }, (err: any) => {
-        done(err);
       });
+
+    assert.strictEqual(i, 4);
   });
 
-  it('correctly handles request that was first 429-throttled and then failed', (done) => {
+  it('correctly handles request that was first 429-throttled and then failed', async () => {
     let i: number = 0;
 
     sinon.stub(_request as any, 'req').callsFake(() => {
       if (i++ === 0) {
-        return Promise.reject({
+        throw {
           response: {
             status: 429,
             headers: {}
           }
-        });
+        };
       }
       else {
-        return Promise.reject('Error');
+        throw 'Error';
       }
     });
```

Comment:
The original file used to have a setTimeout stub here. Why is it missing now? The test may now run a second longer than is necessary because of the retry.

Reply:
See comment above